aptos-move/framework/aptos-framework/sources/aggregator/optional_aggregator.move

```diff
@@ -153,6 +175,17 @@ module aptos_framework::optional_aggregator {
         limit
     }
 
+    /// Adds `value` to optional aggregator, returns `true` on success and `false` on overflow.
+    public fun try_add(optional_aggregator: &mut OptionalAggregator, value: u128): bool {
```

Comment:
I was talking to @georgemitenkov about this earlier today and I think it makes more sense to return more context beyond just a bool. If the intention is to allow devs to handle failures, there's not much they can do if they don't know whether the error is ignorable. Overflow/underflow errors should be clear. So we should return either an enum here (error code) or a proper Result-like struct with more context when there's a failure

Reply:
After another look at this, I think there is really no extra information we can pass out here from these `try_` methods. As @igor-aptos pointed out, we would have:
```
if (aggregator::try_add(x, 5)) {
   ... 
} else {
   ... 
}
```
as equivalent to:
```
if (x.value < x.limit - 5) {
   x.value += 5;
   ...
} else {
   ...
}
```
The only information that might be useful would be the knowing by how much the value was overflowing. But if we detect an overflow, read becomes cheap, so a pattern like this is acceptable?
```
if (aggregator::try_add(x, 5)) {
   ... 
} else {
   let v = aggregator::read(x);
   let delta = aggregator::limit(x) - v;
   ...
}
```
What do you think @movekevin ?

Reply:
Can we guarantee now and in the future that the only error that can arise from try_add is overflow?

Reply:
that's a fair point, but yes, as far as the integer aggregator goes as an abstract state machine, and an interface that is trying to add and subtract and tolerate failures, these APIs and corresponding failure handling everywhere are specifically intended and designed for overflow/underflow. 
I can't imagine any future things that we would want to shoehorn here, or any existing things for that matter (storage error for example is not something that trying to add is ever going to catch). Also, as @igor-aptos also noted, we can always add interfaces, try_add is clearly specific to the addition.

Reply:
Gotcha makes sense